gobblin-audit/src/main/java/gobblin/audit/values/auditor/ValueAuditGenerator.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.audit.values.auditor;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import org.apache.avro.generic.GenericRecord;
+
+import com.typesafe.config.Config;
+
+import gobblin.audit.values.policy.column.ColumnProjectionPolicy;
+import gobblin.audit.values.policy.column.DefaultColumnProjectionPolicyFactory;
+import gobblin.audit.values.policy.row.DefaultRowSelectionPolicyFactory;
+import gobblin.audit.values.policy.row.RowSelectionPolicy;
+import gobblin.audit.values.sink.AuditSink;
+import gobblin.audit.values.sink.DefaultAuditSinkFactory;
+
+
+/**
+ * The class that implements value based auditing. The class captures the values of certain
+ * columns from the rows in the dataset using the {@link ColumnProjectionPolicy}.
+ * This is done for every row or for a sample of the rows as defined by the {@link RowSelectionPolicy}.
+ * The selected rows are then written to the {@link AuditSink}
+ *
+ * {@link ValueAuditGenerator#audit(GenericRecord)} is the method that audits an inputRecord.
+ */
+@AllArgsConstructor
+@Getter
+public class ValueAuditGenerator implements Closeable {
+
+  private final ColumnProjectionPolicy columnProjectionPolicy;
+  private final RowSelectionPolicy rowSelectionPolicy;
+  private final AuditSink auditSink;
+
+  /**
+   * Factory method to create a new {@link ValueAuditGenerator}
+   * @param config job configs
+   * @param runtimeAuditMetadata is used to pass the table specific runtime information like tablename, databaseName, snapshotName etc.
+   *      See {@link ValueAuditRuntimeMetadata}
+   * @return a new {@link ValueAuditGenerator}
+   */
+  public static ValueAuditGenerator create(Config config, ValueAuditRuntimeMetadata runtimeAuditMetadata) {
+    ColumnProjectionPolicy columnProjectionPolicy = DefaultColumnProjectionPolicyFactory.getInstance().create(
+        config.getConfig(DefaultColumnProjectionPolicyFactory.CONFIG_SCOPE_PREFIX),runtimeAuditMetadata.getTableMetadata());
```

Comment:
minor: The scope prefix in the next three cases should be declared in this class not the factories as a different class may want to use a different scope.